### ISSUE TITLE
Debugging exception thrown when executing SmartStep

### ISSUE THREAD

**Issue by teohhanhui** [KnpLabs/FriendlyContexts#164]
There should probably be a debug/verbose output so that we can see what's going on inside a SmartStep (don't swallow all the output from `echo`)

**Comment by teohhanhui** [comment]
@Einenlum What changed?

**Comment by Nek-** [comment]
I'm not sure to get what you mean @teohhanhui.

This is an RFC to have a better support of exceptions that happen inside a SmartStep?

**Comment by Einenlum** [comment]
Actually I closed it but I should have asked @Nek- since he's the main maintainer of this repo now.

I closed it cause it looks like an RFC without concrete proposition, and I'm not sure it's really needed, especially since smartStep seems like a feature that should be deprecated to me.

**Comment by Nek-** [comment]
Legit.

(and it's ok if you close issues, you're collaborator too 😮  )

**Comment by teohhanhui** [comment]
> This is an RFC to have a better support of exceptions that happen inside a SmartStep?

Yeah, because SmartStep is almost completely silent. It swallows everything.

**Comment by Nek-** [comment]
Can you please reformat the issue so we don't get lost anymore? :)

We're going to investigate this. It's not really a bug but I get it's a problem (but for now there is some emergency like "omg we're not compatible with Symfony 3").

**Comment by teohhanhui** [comment]
It's not really an issue for me at the moment, so I'm just gonna leave this as is for now.
